neuralprophet/forecaster.py

```diff
@@ -1878,6 +1957,35 @@ def plot_parameters(
             if quantile not in self.config_train.quantiles:
                 raise ValueError("Selected quantile is not specified in the model configuration.")
 
+        # Check if list is passed and ensure lower case input
+        if components is not None:
+            if type(components) is not list:
+                components = [components]
+            components = [comp.lower() for comp in components]
+
+            # # Error if selected plotting panels are not configured in model or mis-spelled
+            for comp in components:
+                if (
+                    ("trend" in comp and not self.config_trend.n_changepoints > 0)
+                    or ("seasonality" in comp and self.model.config_season is None)
+                    or ("autoregression" in comp and not self.model.n_lags > 0)
+                    or ("lagged_regressors" in comp and self.model.config_lagged_regressors is None)
+                    or ("future_regressors" in comp and self.config_regressors is None)
+                    or ("country_holidays" in comp and self.model.config_country_holidays is None)
+                    or ("events" in comp and self.config_events is None)
+                ):
+                    raise ValueError(" Selected component(s) for plotting is not specified in the model configuration.")
+                elif comp not in [
+                    "trend",
+                    "seasonality",
+                    "autoregression",
+                    "lagged_regressors",
+                    "future_regressors",
+                    "country_holidays",
+                    "events",
+                ]:
+                    raise ValueError(" Selected component(s) for plotting are mis-spelled.")
+
```

Comment:
Can't this validation be extracted to some helper function as we obviously have it twice (very prone to forgetting changes in the other copy)?